src/workerd/io/container-client.c++

```diff
@@ -0,0 +1,277 @@
+
+    auto req = httpClient->request(method, endpoint, headers, requestBody.size());
+    {
+      auto body = kj::mv(req.body);
+      co_await body->write(requestBody.asBytes());
+    }
+    auto response = co_await req.response;
+    auto result = co_await response.body->readAllText();
+    co_return Response{.statusCode = response.statusCode, .body = kj::mv(result)};
+  } else {
+    auto req = httpClient->request(method, endpoint, headers);
+    { auto body = kj::mv(req.body); }
+    auto response = co_await req.response;
+    auto result = co_await response.body->readAllText();
+    co_return Response{.statusCode = response.statusCode, .body = kj::mv(result)};
+  }
+}
+
+kj::Promise<bool> ContainerClient::isContainerRunning() {
+  // Docker API v1.50: GET /containers/{id}/json
+  auto endpoint = kj::str("/containers/", containerName, "/json");
+
+  auto response = co_await dockerApiRequest(kj::HttpMethod::GET, endpoint);
+  if (response.statusCode == 404) {
+    co_return false;
+  }
+  JSG_REQUIRE(response.statusCode == 200, Error, "Container inspect failed");
+
+  // Parse JSON response
+  capnp::JsonCodec codec;
+  codec.handleByAnnotation<docker_api::Docker::ContainerInspectResponse>();
+  capnp::MallocMessageBuilder message;
+  auto jsonRoot = message.initRoot<docker_api::Docker::ContainerInspectResponse>();
+  codec.decode(response.body, jsonRoot);
+
+  // Look for Status field in the JSON object
+  if (jsonRoot.hasState()) {
+    auto state = jsonRoot.getState();
+    if (state.hasStatus()) {
+      auto status = state.getStatus();
+      co_return status == "running";
+    }
+  }
+  co_return false;
+}
+
+kj::Promise<void> ContainerClient::createContainer(
+    kj::Maybe<capnp::List<capnp::Text>::Reader> entrypoint,
+    kj::Maybe<capnp::List<capnp::Text>::Reader> environment) {
+  // Docker API v1.50: POST /containers/create
+  capnp::MallocMessageBuilder message;
+  auto jsonRoot = message.initRoot<docker_api::Docker::ContainerCreateRequest>();
+  jsonRoot.setImage(imageTag);
+  // Add entrypoint if provided
+  KJ_IF_SOME(ep, entrypoint) {
+    auto jsonCmd = jsonRoot.initCmd(ep.size());
+    for (uint32_t i = 0; i < ep.size(); i++) {
+      jsonCmd.set(i, ep[i]);
+    }
+  }
+
+  // Add environment variables if provided
+  KJ_IF_SOME(env, environment) {
+    auto jsonEnv = jsonRoot.initEnv(env.size());
+    for (uint32_t i = 0; i < env.size(); i++) {
+      jsonEnv.set(i, env[i]);
+    }
+  }
+
+  // Encode to JSON string
+  capnp::JsonCodec codec;
+  codec.handleByAnnotation<docker_api::Docker::ContainerInspectResponse>();
+  kj::String jsonBody = codec.encode(jsonRoot);
+
+  auto endpoint = kj::str("/containers/create?name=", containerName);
+
+  auto response = co_await dockerApiRequest(kj::HttpMethod::POST, endpoint, jsonBody.asPtr());
+
+  // statusCode 201 refers to "container created successfully"
+  // statusCode 409 refers to "conflict"
+  if (response.statusCode != 201 && response.statusCode != 409) {
```

Comment:
why `response.statusCode != 409` ?
sounds like an error from the comment?

Reply:
Oh got what you mean...
If a container already exists it's not an error because you will re-use it.
Is that it?

Maybe add a comment for people that are slow to understand as I am ;)

But what if the env variables or command is different?